extensions/parquet/table/src/main/java/io/deephaven/parquet/table/location/ParquetTableLocation.java

```diff
@@ -692,90 +738,144 @@ private void iterateRowGroupsAndRowSet(final RowSet input, final RowGroupAndRowS
-
-        // Numeric values need no conversion
-        if (NumericTypeUtils.isIntegralOrChar(clazz) || NumericTypeUtils.isFloat(clazz)) {
-            return new Pair<>(statistics.genericGetMin(), statistics.genericGetMax());
-        }
-
-        if (statistics.type().getLogicalTypeAnnotation() == stringType()) {
-            return new Pair<>(statistics.minAsString(), statistics.maxAsString());
+    private static long getNullCount(final Statistics<?> statistics) {
+        if (statistics == null || statistics.isEmpty() || !statistics.isNumNullsSet()) {
+            return -1L;
         }
-
-        if (statistics.type()
-                .getLogicalTypeAnnotation() instanceof LogicalTypeAnnotation.DecimalLogicalTypeAnnotation) {
-            final int scale = ((LogicalTypeAnnotation.DecimalLogicalTypeAnnotation) statistics.type()
-                    .getLogicalTypeAnnotation()).getScale();
-            if (scale == 0) {
-                // Return the min and max values as BigInteger
-                return new Pair<>(new BigInteger(statistics.getMinBytes()), new BigInteger(statistics.getMaxBytes()));
-            } else {
-                // We need to convert the min and max values to BigDecimal
-                return new Pair<>(new BigDecimal(new BigInteger(statistics.getMinBytes()), scale),
-                        new BigDecimal(new BigInteger(statistics.getMaxBytes()), scale));
-            }
-        }
-        return null;
+        return statistics.getNumNulls();
     }
 
     /**
-     * Apply the range filter to the row groups and return the result.
+     * Apply the filter to the row group metadata and return the result.
      */
     @NotNull
-    private PushdownResult pushdownRangeFilter(
-            final AbstractRangeFilter rf,
+    private PushdownResult pushdownRowGroupMetadata(
+            final WhereFilter filter,
             final List<Integer> columnIndices,
             final PushdownResult result) {
         final RowSetBuilderSequential maybeBuilder = RowSetFactory.builderSequential();
         final MutableLong maybeCount = new MutableLong(0);
 
-        // Only one column in a RangeFilter
+        // Only one column in these filters
         final Integer columnIndex = columnIndices.get(0);
+
         final List<BlockMetaData> blocks = parquetMetadata.getBlocks();
         iterateRowGroupsAndRowSet(result.maybeMatch(), (rgIdx, rs) -> {
-            final Pair<Object, Object> p = getMinMax(
-                    blocks.get(rgIdx).getColumns().get(columnIndex).getStatistics());
+            final Statistics<?> statistics = blocks.get(rgIdx).getColumns().get(columnIndex).getStatistics();
+
+            final Optional<MinMax<?>> minMaxFromStatistics = MinMaxFromStatistics.get(statistics);
+            final long nullCount = getNullCount(statistics);
 
-            if (p == null || rf.overlaps(p.first, p.second)) {
+            if (minMaxFromStatistics.isEmpty() || nullCount < 0) {
```

Comment:
It doesn't follow that an unknown null count means we can't e.g. determine if there is an overlap for a match filter or most range filters.  It may be that we don't actually care, because if we have one we are likely to have the other.